### PR TITLE
Rename `instance()`->`instantiate()` when it's a verb

### PR DESCRIPTION
**Pull request by AaronRecord**
See https://github.com/godotengine/godot/issues/16863#issuecomment-367061984

### CODE REVIEW

**Comment by pycbouh** [comment]
Shouldn't there be a change in the documentation as well?

**Comment by Xrayez** [comment]
I'm not particularly against the rename, but it's not only about simple refactor thing. The entire codebase should be reviewed for usage of "instance" word, including comments.

File: scene/resources/packed_scene.h [hunk 1]
```diff
@@ -137,7 +137,7 @@ class SceneState : public RefCounted {
 	void clear();
 
 	bool can_instance() const;
```
Comment (Xrayez) [inline scene/resources/packed_scene.h:139]: For instance:

```suggestion
	bool can_instantiate() const;
```

But can also rename rename this to `is_instantiable()`, depending on the context.

**Comment by pycbouh** [comment]
> The existing "instance" word sounds fine to me (and is easier to type), but I'm not a native speaker. English allows to use nouns as verbs, so that's why it probably sounds natural to me

"Instance" can actually be used as a verb, but it has an unrelated meaning.

**Comment by LightningAA** [comment]
I think `instantiate()` makes more sense  than `instance()` for meaning "create an instance of ...".

**Comment by fire** [comment]
I can see myself typoing a bit with this compared to instance.

Would mistakenly rearrange the voxels.

**Comment by LightningAA** [comment]
Should I open a proposal for this? The comment has +24 -0  likes/dislikes

**Comment by LightningAA** [comment]
> I can see myself typoing a bit with this compared to instance.
> 
> Would mistakenly rearrange the vowels.

Well ideally that's what autocomplete and compiler errors are for, instance and instantiate are the same for the first 6 letters.

**Comment by akien-mga** [comment]
Approved in PR review meeting today. The documentation needs to be synced with `--doctool` (and ensuring that the descriptions for the renamed methods don't get lost).

**Comment by LightningAA** [comment]
(rebased)

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Needs another rebase as I merged a bunch of PR (+ a couple might have added new calls to `instance()` to change).

Then I'll merge this PR first :)

**Comment by LightningAA** [comment]
Okay I think everything *should* be good to go.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**